run-checks

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash -e
+
+# shellcheck disable=SC2145
+
+#----------------------------------------------------#
+# PRINTING ROUTINES                                  #
+#----------------------------------------------------#
+
+P_RESET="\e[0m"
+P_RED="\e[31m"
+P_GREEN="\e[32m"
+
+print_newline() {
+	echo -e "$P_RESET"
+}
+
+print_normal() {
+	echo -e "$P_RESET$@"
+}
+
+print_red() {
+	echo -e "$P_RED$@$P_RESET"
+}
+
+print_green() {
+	echo -e "$P_GREEN$@$P_RESET"
+}
+
+print_special() {
+	echo -ne "$P_RESET$@"
+}
+
+print_err() {
+	print_newline
+	pr "Error: $1"
+	print_newline
+	exit 1
+}
+
+#----------------------------------------------------#
+# TEST OUTPUT CONTROL                                #
+#----------------------------------------------------#
+
+print_test() {
+	print_special "$@"
+}
+
+run() {
+        if $VERBOSE; then
+                v=$(exec 2>&1 && set -x && set -- "$@")
+		echo
+                echo "#${v#*--}"
+		set +e
+                "$@"
+		RET=$?
+		set -e
+        else
+		set +e
+                "$@" >/dev/null 2>&1
```

Comment:
I'm not sure about the benefit of this, why would we ever want to run this in non-verbose mode? The experience is something like:
```
$ ./run-checks 
Check Build ... Pass
Check Vet ... Pass
Check Formatting ... 
Error: Formatting check failed. (run 'run-checks --verbose' for details)
```
and then it will be re-run with --verbose.

Reply:
Its a magic piece of code I use in build scripts to control verbosity. As you suggest, useless in a test script. I made the default behaviour much nicer, and removed this piece of code.